src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx

```diff
@@ -122,33 +121,46 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
         [setHoverNode, updateHighlight, highlightLinks, highlightNodes]
     );
 
+    let clickedNodeKey = null;
+    if (clickedNode !== null) {
+        clickedNodeKey = clickedNode.id;
+    }
+
+    let hoverNodeKey = null;
+    if (hoverNode !== null) {
+        hoverNodeKey = (hoverNode as any).id;
+    }
+
```

Comment:
Add a comment here, something like "We only want to rerender when the id of a node changes, but we don't want to update based on any of its other attributes"